### PR TITLE
rewrote the file reading so that the entire file is read into memory …

### PR DESCRIPTION
**Pull request by jgriesfeller**
so that there's one place to determine if the file is UTF-8 or UTF-16 encoded or a complete failure. Saves a lot of if statements for the price of temporary RAM usage.

### CODE REVIEW

**Comment by jgliss** [comment]
Does this make #459 obsolete,  @jgriesfeller and @dulte?

**Comment by dulte** [comment]
Yes, this makes the previous one obsolete. I can remove it

File: pyaerocom/io/read_eea_aqerep_base.py [hunk 1]
```diff
@@ -455,9 +448,9 @@ def _read_metadata_file(self, filename=None):
             bad_line_arr = []
             for line in f:
                 rows = line.rstrip().split('\t')
-
-                if len(rows) < 24:
```
Comment (jgriesfeller) [inline pyaerocom/io/read_eea_aqerep_base.py:459]: removed constant usage to make the code more roibust

**Review by dulte — APPROVED** [review]
Looks good

**Pull request merged**